test/ResultTest.cpp

```diff
@@ -439,147 +405,107 @@ TEST(Result, verifyApplyLimitOffsetHandlesZeroLimitCorrectly) {
-    EXPECT_THROW(result.checkDefinedness(map), ad_utility::Exception);
-  }
-  {
-    Result result{wrongTable3.clone(), {}, LocalVocab{}};
-    EXPECT_THROW(result.checkDefinedness(map), ad_utility::Exception);
-  }
-  for (auto& generator : getAllSubSplits(correctTable1)) {
-    Result result{std::move(generator), {}, LocalVocab{}};
-    result.checkDefinedness(map);
-    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
-  }
-  for (auto& generator : getAllSubSplits(correctTable2)) {
-    Result result{std::move(generator), {}, LocalVocab{}};
-    result.checkDefinedness(map);
-    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
-  }
-  for (auto& generator : getAllSubSplits(correctTable3)) {
-    Result result{std::move(generator), {}, LocalVocab{}};
-    result.checkDefinedness(map);
-    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
-  }
-  for (auto& generator : getAllSubSplits(correctTable4)) {
+  for (auto& generator : getAllSubSplits(*GetParam())) {
     Result result{std::move(generator), {}, LocalVocab{}};
     result.checkDefinedness(map);
     EXPECT_NO_THROW(consumeGenerator(result.idTables()));
   }
-  for (auto& generator : getAllSubSplits(wrongTable1)) {
-    Result result{std::move(generator), {}, LocalVocab{}};
-    result.checkDefinedness(map);
-    EXPECT_THROW(consumeGenerator(result.idTables()), ad_utility::Exception);
+}
+
+INSTANTIATE_TEST_SUITE_P(SuccessCases, ResultDefinednessTestS,
+                         testing::Values(&correctTable1, &correctTable2,
+                                         &correctTable3, &correctTable4));
+
+// _____________________________________________________________________________
+TEST_P(ResultDefinednessTestF,
+       verifyCheckDefinednessDoesThrowIfColumnIsNotDefinedWhenClaimingItIs) {
+  if constexpr (!ad_utility::areExpensiveChecksEnabled) {
+    GTEST_SKIP_("Expensive checks are disabled, skipping test.");
   }
-  for (auto& generator : getAllSubSplits(wrongTable2)) {
-    Result result{std::move(generator), {}, LocalVocab{}};
-    result.checkDefinedness(map);
-    EXPECT_THROW(consumeGenerator(result.idTables()), ad_utility::Exception);
+  VariableToColumnMap map{
+      {Variable{"?a"}, {0, ColumnIndexAndTypeInfo::AlwaysDefined}},
+      {Variable{"?b"}, {1, ColumnIndexAndTypeInfo::PossiblyUndefined}}};
+
+  {
+    Result result{GetParam()->clone(), {}, LocalVocab{}};
+    EXPECT_THROW(result.checkDefinedness(map), ad_utility::Exception);
   }
-  for (auto& generator : getAllSubSplits(wrongTable3)) {
+  for (auto& generator : getAllSubSplits(*GetParam())) {
     Result result{std::move(generator), {}, LocalVocab{}};
     result.checkDefinedness(map);
     EXPECT_THROW(consumeGenerator(result.idTables()), ad_utility::Exception);
   }
 }
+
+INSTANTIATE_TEST_SUITE_P(FailureCases, ResultDefinednessTestF,
+                         testing::Values(&wrongTable1, &wrongTable2,
+                                         &wrongTable3));
```

Comment:
I am wondering, 
If you can at least merge the respective
`success` and `failure` cases by a common outer implementation
(maybe template them on what to do with the actual IdTables).
But again this is already much better.

Reply:
I could provide a "should throw/should not throw" bool along with the values but I doubt this helps readability.  It would just create a an if case each around both assertions. Parameterized tests in gtest are so unnecessarily clunky to use unfortunately.